src/main/steps/common/yourhearings/hearings/utils.ts

```diff
@@ -0,0 +1,168 @@
+  } else {
+    partyData =
+      partyType === PartyType.RESPONDENT ? req.session.userCase.respondentsFL401 : req.session.userCase.applicantsFL401;
+    hearingMethod = attendees.find(attendee => attendee.partyID === partyData.partyId)!.hearingSubChannel;
+  }
+  if (hearingMethod !== null) {
+    const a = {
+      vid: [
+        HEARING_METHOD.VID,
+        HEARING_METHOD.VIDCVP,
+        HEARING_METHOD.VIDOTHER,
+        HEARING_METHOD.VIDPVL,
+        HEARING_METHOD.VIDSKYPE,
+        HEARING_METHOD.VIDTEAMS,
+        HEARING_METHOD.VIDVHS,
+      ],
+      tel: [
+        HEARING_METHOD.TEL,
+        HEARING_METHOD.TELBTM,
+        HEARING_METHOD.TELCVP,
+        HEARING_METHOD.TELOTHER,
+        HEARING_METHOD.TELSKYP,
+        HEARING_METHOD.TELTEMP,
+      ],
+      inter: [HEARING_METHOD.INTER],
+      na: [HEARING_METHOD.NA],
+      onpprs: [HEARING_METHOD.ONPPRS],
+    };
+
+    return lang[Object.keys(a).find(key => a[key].includes(hearingMethod)) ?? hearingMethod];
+  }
+  return '';
+}
+
+export const generateHearingDaySchedule = (
+  hearing: HearingsList,
+  req: AppRequest<Partial<Case>>,
+  hearingDays: hearingDay[],
+  hearingMethod
+) => {
+  for (const schedule of hearing.hearingDaySchedule!) {
+    const startDate = schedule.hearingStartDateTime!;
+    const formattedDate = new Date(startDate);
+    const amPm = formattedDate.getHours() < 12 ? 'am' : 'pm';
+    const endDate = new Date(schedule.hearingEndDateTime!);
+    const hearingDate = dateGenarationWithDay(req, startDate);
+    const startTime = getProperTime(formattedDate);
+    const lang = req.session.lang === 'cy' ? cy : en;
+    const preparedStartTime =
+      hearingMethod === lang.inter ? startTime + ' ' + amPm + '<br>' + lang.inPersonTime : startTime + ' ' + amPm;
+    const diff = Math.abs(formattedDate.valueOf() - endDate.valueOf()) / 1000;
+    const durationInDayOrHours = Math.floor(diff / 3600) % 24;
+    const minutes = Math.floor(diff / 60) % 60;
+    const hearingDurationText = generateHearingText(durationInDayOrHours, minutes, lang);
+
+    const judgeName = schedule.hearingJudgeName;
+    const venue = schedule.hearingVenueName;
+    const address = schedule.hearingVenueAddress;
+    const roomId = schedule.hearingRoomId;
+    const detailsBlock: Row[] = [];
+    if (hearingMethod === lang.inter) {
+      detailsBlock.push(
+        { key: lang.venue, value: venue },
+        { key: lang.address, value: address },
+        { key: lang.roomId, value: roomId }
+      );
+    } else if (hearingMethod === lang.tel || lang.vid) {
+      detailsBlock.push({ key: lang.hearingLinkHeading, value: lang.hearingLink });
+    }
+
+    hearingDays.push({
+      hearingDate,
+      startTime,
+      amPm,
+      preparedStartTime,
+      durationInDayOrHours,
+      minutes,
+      hearingDurationText,
+      judgeName,
+      venue,
+      address,
+      roomId,
+      detailsBlock,
+    });
+  }
+};
+
+export function getProperTime(date: Date): string {
```

Comment:
what does it mean by getProperTime. Rename it to generateHearingTimeDisplayText

Reply:
done